src/component/movieList.js

```diff
@@ -0,0 +1,107 @@
+import { Typography } from '@mui/material';
+import React, { useState } from 'react';
+import Rating from '@mui/material/Rating';
+import { useEffect } from 'react';
+import StarIcon from '@mui/icons-material/Star';
+import Link from 'next/link';
+
+const MoviesList = ({ records }) => {
+  const [hoveredMovie, setHoveredMovie] = useState(null);
+
+  const gridStyle = {
```

Comment:
It would be more modular and maintainable if we abstract the CSS styling into a separate css stylesheet alongside this component.